pkg/api/targetgroup_deployment.go

```diff
@@ -0,0 +1,122 @@
+package api
+
+import (
+	"net/http"
+
+	"github.com/common-fate/apikit/apio"
+	"github.com/common-fate/common-fate/pkg/storage"
+	"github.com/common-fate/common-fate/pkg/targetgroup"
+	"github.com/common-fate/common-fate/pkg/types"
+)
+
+// Your GET endpoint
+// (GET /api/v1/target-group-deployments)
+func (a *API) ListTargetGroupDeployments(w http.ResponseWriter, r *http.Request) {
+
+	ctx := r.Context()
+	res := types.ListTargetGroupDeploymentsResponse{
+		Res: []types.TargetGroupDeployment{},
+	}
+
+	listTargetGroupDeployments := storage.ListTargetGroupDeployments{}
+
+	dbq, err := a.DB.Query(ctx, &listTargetGroupDeployments)
+	if err != nil {
+		apio.Error(ctx, w, err)
+		return
+	}
+	if dbq.NextPage != "" {
+		res.Next = dbq.NextPage
+	}
+	for _, r := range listTargetGroupDeployments.Result {
+		res.Res = append(res.Res, r.ToAPI())
+	}
+
+	apio.JSON(ctx, w, res, http.StatusOK)
+
+}
+
+// (POST /api/v1/target-group-deployments)
+func (a *API) CreateTargetGroupDeployment(w http.ResponseWriter, r *http.Request) {
+
+	ctx := r.Context()
+
+	var b types.CreateTargetGroupDeploymentRequest
+	err := apio.DecodeJSONBody(w, r, &b)
+	if err != nil {
+		apio.Error(ctx, w, err)
+		return
+	}
+
+	// TODO: run pre-lim checks to ensure aws account/arn are valid
+
+	dbInput := targetgroup.Deployment{
+		ID:           types.NewTargetGroupDeploymentID(),
+		FunctionARN:  b.FunctionArn,
+		Runtime:      b.Runtime,
+		AWSAccount:   b.AwsAccount,
+		Healthy:      false,
+		Diagnostics:  []targetgroup.Diagnostic{},
```

Comment:
The default status for a new deployment is unhealthy. This is because the deployment has never been healthchecked.

A deployment which cannot be reached should have a Diagnostic log that says it is not online.

Define this diagnostic log in the targetgroup package and use it here

```suggestion
		Diagnostics:  []targetgroup.Diagnostic{},
```